### PR TITLE
Handling policy renewals from mobile

### PR DESCRIPTION
**Pull request by tdethier**
This PR changes a few things for handling mobile policy renewals through GraphQL:
- New query for listing the PolicyRenewals objects
- Update to the service handling renewals: now, it's also looking for any PolicyRenewals objects and archives them
- A mutation for creating PolicyRenewals objects was created in case we might need it later

### CODE REVIEW

**Review by malinowskikam — APPROVED** [review]

**Comment by malinowskikam** [comment]
Brought this PR up to date as part of [OP-2038](https://openimis.atlassian.net/browse/OP-2038)

[OP-2038]: https://openimis.atlassian.net/browse/OP-2038?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

File: policy/gql_mutations.py [hunk 1]
```diff
@@ -50,6 +50,20 @@ def do_mutate(cls, perms, user, **data):
...
+            logger.info("Deleting the optional PolicyRenewals after renewing")
+            previous_policy = (Policy.objects.filter(validity_to__isnull=True,
+                                                     family_id=data["family_id"],
+                                                     product_id=data["product_id"],
+                                                     status__in=[Policy.STATUS_EXPIRED, Policy.STATUS_ACTIVE])
+                                             .order_by("-id")
+                                             .first())
+            if not previous_policy:
+                logger.error("Can't find the policy that was renewed - not deleting the PolicyRenewals")
+            policy_renewals = PolicyRenewal.objects.filter(policy=previous_policy, validity_to__isnull=True)
```
Comment (dragos-dobre) [inline policy/gql_mutations.py:64]: Should it return something before if there is no previous policy ?

**Review by dragos-dobre — CHANGES_REQUESTED** [review]

File: policy/gql_mutations.py (hunk 1 above)
Comment (malinowskikam) [reply policy/gql_mutations.py:64]: It depends. It's happening after the policy renewal is created so they just igrone if there is no previous policy.Should that cause the whore renewal to be rejected?

Comment (dragos-dobre) [reply policy/gql_mutations.py:64]: Let's keep it as such for now and we'll wait for implementation request to change this.

**Review by dragos-dobre — APPROVED** [review]

**Pull request merged**